crates/fuel-streams-publisher/src/inputs.rs

```diff
@@ -60,11 +56,12 @@ fn packet_from_input(
             let contract_id = contract.contract_id;
             PublishPacket::new(
                 input,
-                InputsContractSubject::new()
-                    .with_tx_id(Some(tx_id))
-                    .with_index(Some(index))
-                    .with_contract_id(Some(contract_id.into()))
-                    .arc(),
+                InputsContractSubject::build(
+                    Some(tx_id),
+                    Some(index),
+                    Some(contract_id.into()),
+                )
```

Comment:
Using this `build` API could introduce quite a lot of hidden bugs, that's why we should prefer either using the `with_*` APIs or, if we have performance concerns, use the subject struct directly `InputContractSubject { tx_id: ,...}`

Reply:
Yeah, my main concern here was mainly performance, because it's just one call instead of many, and the `with_*` creates a new instance on each call. But yeah, go directly with the struct is a better approach.

Reply:
Changed @Jurshsmith